lang/py/avro/io.py

```diff
@@ -222,7 +222,10 @@ def read(self, n: int) -> bytes:
         """
         Read n bytes.
         """
-        return self.reader.read(n)
+        read_bytes = self.reader.read(n)
```

Comment:
You mentioned that the original `avro-python3` code also checked that n is non-negative.  I think that's a good idea too!

Reply:
Added, raises `InvalidAvroBinaryEncoding`.